e2e/wsl-integrations.e2e.spec.ts

```diff
@@ -5,156 +5,146 @@
+import { spawn, spawnFile } from '@pkg/utils/childProcess';
+import { Log } from '@pkg/utils/logging';
 
 import type { ElectronApplication, Page } from '@playwright/test';
 
+type WSLFixtures = {
+  wslError: string | undefined;
+};
+
+const console = new Log(path.basename(__filename, '.ts'), reportAsset(__filename, 'log'));
+
+const test = testBase.extend<WSLFixtures>({
+  // eslint-disable-next-line no-empty-pattern -- skip first arg
+  wslError: async({}, use) => {
+    if (process.platform !== 'win32') {
+      await use('Only applies to Windows');
+
+      return;
+    }
+    try {
+      console.log('Checking WSL installation...');
+      // WSL is expected to return with a non-zero exit code here.
+      const buffers: Buffer[] = [];
+      const proc = spawn('wsl.exe', ['--help'], { stdio: ['ignore', 'pipe', await console.fdStream], windowsHide: true });
+
+      proc.stdout.on('data', (data) => {
+        buffers.push(Buffer.isBuffer(data) ? data : Buffer.from(data));
+      });
+      await new Promise<void>((resolve, reject) => {
+        proc.on('exit', (code, signal) => {
+          if (signal) {
+            reject(`signal ${ signal }`);
+          } else {
+            resolve();
+          }
+        });
+        proc.on('error', reject);
+      });
+
+      const textOut = Buffer.concat(buffers).toString('utf16le');
+
+      if (textOut.includes('--exec')) {
+        console.debug('WSL help output contains --exec');
+        await use(undefined);
+      } else {
+        console.debug('WSL is not installed');
+        await use('WSL is not installed');
+      }
+    } catch (ex) {
+      console.log(`Failed to check WSL: ${ ex }`);
+      await use(`Failed to check WSL: ${ ex }`);
+    }
+  },
+});
+
 test.describe('WSL Integrations', () => {
   test.describe.configure({ mode: 'serial' });
-  if (os.platform() !== 'win32') {
-    test.skip();
-  }
+  test.skip(process.platform !== 'win32', 'Only applies on Windows');
 
   /** The directory containing our mock wsl.exe */
   let workdir = '';
   /** The environment variables, before our tests. */
   let electronApp: ElectronApplication;
   let page: Page;
-  let preferencesWindow: Page;
+  let preferencesWindow: PreferencesPage;
+  /** The list of registered distros. */
+  const distros: string[] = [];
+
+  async function getPrefWindow(): Promise<PreferencesPage> {
+    if (preferencesWindow && !preferencesWindow.page.isClosed()) {
+      return preferencesWindow;
+    }
+    await new NavPage(page).preferencesButton.click();
+    const prefPage = await electronApp.waitForEvent('window', page => /preferences/i.test(page.url()));
+
+    // Wait for the window to actually load (i.e. transition from
+    // app://index.html/#/preferences to app://index.html/#/Preferences#general)
+    await prefPage.waitForURL(/Preferences#/i);
+    preferencesWindow = new PreferencesPage(prefPage);
+
+    return preferencesWindow;
+  }
+
+  test.beforeAll(({ wslError }) => {
```

Comment:
Where does the `wslError` argument come from?

Reply:
The fixture, from line 33. (Or, I guess, 26 if you want to talk types).  I do need to update the PR to have a relevant comment here, though.